account_payment_order_validation_move_line/wizard/account_payment_line_create.py

```diff
@@ -0,0 +1,24 @@
+# © 2015-2016 Akretion (<https://www.akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class AccountPaymentLineCreate(models.TransientModel):
+    _inherit = "account.payment.line.create"
+
+    select_for_payment_move_line = fields.Boolean(
+        string="Take only to pay",
+        related="order_id.payment_mode_id.default_selected_for_payment_id",
+        store=True,
+        readonly=False,
+    )
+
+    def _prepare_move_line_domain(self):
+
+        res = super(AccountPaymentLineCreate, self)._prepare_move_line_domain()
+
+        if self.select_for_payment_move_line:
+            res += [("move_id.selected_for_payment", "=", True)]
```

Comment:
I wonder if the condition should be more complicated, because, this selected_for_payment is only meant to be set on invoices, not in regular accounting entry. But this payment order can accept accounting entry that are not invoices, depending on the filter `invoice`.

So I guess the domain as it is now is ok in case `invoice` field is True, but if not, we'd have to have a domain like 
['|', '&', ('move_type', 'in', ("in_invoice", "out_invoice", "in_refund", "out_refund")), ("move_id.selected_for_payment", "=", True), ('move_type', '=', 'entry')]
So I would check
`if self.invoice and self.select_for_payment_move_line`
then add the domain` [("move_id.selected_for_payment", "=", True)]`
`elif self.select_for_payment_move_line`
then add the more complex domain to accept regular accounting entries